File: LeetCode/Python Solutions/Reverse Prefix of Word/reverse.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        # Find the index of the first occurrence of ch
        index = word.find(ch)
        
        # If ch is not found, return the original word
        if index == -1:
            return word
        
        # Reverse the substring from start to index (inclusive) and concatenate with the rest
        return word[:index+1][::-1] + word[index+1:]

File: LeetCode/Java Solutions/Reverse Prefix of Word/reverse.java
public class reverse {
    public String reversePrefix(String word, char ch) {
        // Find the index of the first occurrence of ch
        int index = word.indexOf(ch);
        
        // If ch is not found, return the original word
        if (index == -1) {
            return word;
        }
        
        // Reverse the substring from start to index (inclusive)
        StringBuilder reversed = new StringBuilder(word.substring(0, index + 1));
        reversed.reverse();
        
        // Concatenate the reversed substring with the rest of the word
        return reversed.toString() + word.substring(index + 1);
    }
}
